src/libraries/System.Text.RegularExpressions/tests/FunctionalTests/Regex.Match.Tests.cs

```diff
@@ -19,869 +20,919 @@ public static IEnumerable<object[]> Match_MemberData()
+                yield return (@"\W.+?\D", "seq 012 of 3 digits", options, 0, 19, true, " 012 ");
+                yield return (@"\W.{1,7}?\D", "seq 012 of 3 digits", options, 0, 19, true, " 012 ");
+                yield return (@"\W.{1,2}?\D", "seq 012 of 3 digits", options, 0, 19, true, " of");
+                yield return (@"\W.*?\b", "digits:0123456789", options, 0, 17, true, ":");
+                yield return (@"\B.*?\B", "e.g:abc", options, 0, 7, true, "");
+                yield return (@"\B\W+?", "e.g:abc", options, 0, 7, false, "");
+                yield return (@"\B\W*?", "e.g:abc", options, 0, 7, true, "");
+
+                // While not lazy loops themselves, variants of the prior case that should give same results here
+                yield return (@"\B\W*", "e.g:abc", options, 0, 7, true, "");
+                yield return (@"\B\W?", "e.g:abc", options, 0, 7, true, "");
+
+                //mixed lazy and eager counting
+                yield return ("z(a{0,5}|a{0,10}?)", "xyzaaaaaaaaaxyz", options, 0, 15, true, "zaaaaa");
+            }
+        }
 
-                    yield return ("\uA640", longCharacterRange, options, 0, 0x2000, false, "");
-                    yield return ("[\u3000-\u3001]", longCharacterRange, options, 0, 0x2000, false, "");
-                    yield return ("[\uA640-\uA641][\u3000-\u3010]", longCharacterRange, options, 0, 0x2000, false, "");
+        [OuterLoop("Takes several seconds to run")]
+        [Fact]
+        public async Task AllEnginesProduceTheSameCaptures()
+        {
+            // Some of the match test cases have captures.  For those that do, validate the captures are the same across engines.
 
-                    if (!RegexHelpers.IsNonBacktracking(engine))
-                    {
-                        yield return ("\u1000", longCharacterRange, options | RegexOptions.RightToLeft, 0, 0x2000, true, "\u1000");
-                        yield return ("[\u1000-\u1001]", longCharacterRange, options | RegexOptions.RightToLeft, 0, 0x2000, true, "\u1001");
-                        yield return ("[\u1000][\u1001-\u1010]", longCharacterRange, options, 0, 0x2000, true, "\u1000\u1001");
+            (string Pattern, string Input, RegexOptions Options, int Beginning, int Length, bool ExpectedSuccess, string ExpectedValue)[] interpreterTests =
+                Match_MemberData_Cases(RegexEngine.Interpreter)
+                .Where(i => i.ExpectedSuccess)
+                .ToArray();
 
-                        yield return ("\uA640", longCharacterRange, options | RegexOptions.RightToLeft, 0, 0x2000, false, "");
-                        yield return ("[\u3000-\u3001][\uA640-\uA641]", longCharacterRange, options | RegexOptions.RightToLeft, 0, 0x2000, false, "");
-                    }
-                }
+            (RegexEngine e, HashSet<(string Pattern, string Input, RegexOptions Options, int Beginning, int Length, bool ExpectedSuccess, string ExpectedValue)>)[] otherTests =
```

Comment:
Should this filter out interpreter engine so that we don't compare interpreted vs interpreted?

Reply:
Probably :)